lib/sentry/context.ex

```diff
@@ -1,78 +1,81 @@
 defmodule Sentry.Context do
   @moduledoc """
-    Provides a method to store user, tags, and extra context when an
-    event is reported. The contexts will be fetched and merged into
-    the event.
+  Provides a method to store user, tags, and extra context when an
+  event is reported. The contexts will be fetched and merged into
+  the event.
 
-    Sentry.Context uses the Process Dictionary to store this state.
-    This imposes some limitations. The state will only exist within 
-    the current process, and the context will die with the process.
+  Sentry.Context uses the Process Dictionary to store this state.
+  This imposes some limitations. The state will only exist within
+  the current process, and the context will die with the process.
 
-    For example, if you add context inside your controller and an 
-    error happens in a Task the context won't send.
+  For example, if you add context inside your controller and an
+  error happens in a Task the context won't send.
   """
+
   @process_dictionary_key :sentry_context
   @user_key :user
   @tags_key :tags
   @extra_key :extra
   @request_key :request
   @breadcrumbs_key :breadcrumbs
 
-  def get_all do
-    context = get_context()
-    %{
-      user: Map.get(context, @user_key, %{}),
-      tags: Map.get(context, @tags_key, %{}),
-      extra: Map.get(context, @extra_key, %{}),
-      request: Map.get(context, @request_key, %{}),
-      breadcrumbs: Map.get(context, @breadcrumbs_key, []),
-    }
-  end
+  @keys [{:user, @user_key},
+         {:tags, @tags_key},
+         {:extra, @extra_key},
+         {:http, @request_key}]
 
-  def set_extra_context(map) when is_map(map) do
-    get_context()
-    |> set_context(@extra_key, map)
-  end
+  @type data :: keyword() | map()
 
-  def set_user_context(map) when is_map(map) do
-    get_context()
-    |> set_context(@user_key, map)
-  end
+  def get_all do
+    context = get_context()
 
-  def set_tags_context(map) when is_map(map) do
-    get_context()
-    |> set_context(@tags_key, map)
+    for {_, key} <- @keys, into: %{} do
+      {key, Map.get(context, key, %{})}
+    end
+    |> Map.put(@breadcrumbs_key, Enum.reverse(Map.get(context, :breadcrumbs, [])))
   end
 
-  def set_http_context(map) when is_map(map) do
-    get_context()
-    |> set_context(@request_key, map)
+  Enum.each @keys, fn {name, key} ->
+    @spec unquote(:"set_#{name}_context")(data):: any()
+    def unquote(:"set_#{name}_context")(map) do
+      map
+      |> Map.new
```

Comment:
I wanted to add 3rd element in tuples that would be `default` and use `Enum.into(unquote default)` but hit the wall with `%{}`. I need to investigate it more.